src/main/java/hsien/Hsien.java

```diff
@@ -0,0 +1,260 @@
+    }
+
+    public static void readFile(ArrayList<Task> messages) {
+        try {
+            File f = new File("tasks.txt");
+            Scanner line = new Scanner(f);
+            while (line.hasNext()) {
+                String[] processedLine = processFileLine(line.nextLine().replace("\n", ""));
+                addTaskFromFile(processedLine[0], processedLine[1], processedLine[2], messages);
+            }
+        } catch (FileNotFoundException e) {
+            // Handle the case where the file does not exist
+            System.out.println("File not found");
+        } catch (HsienException e) {
+            System.out.println("Incorrect file input");
+        }
+    }
+
+    public static void writeFile(ArrayList<Task> messages) {
+        try {
+            FileWriter fw = new FileWriter("tasks.txt");
+            for (Task task: messages) {
+                fw.write(task.getStatusDescription() + "\n");
+            }
+            fw.close();
+        } catch (IOException e) {
+            System.out.println("An error occurred while writing to the file.");
+        }
+    }
+
+    public static String[] processCommand(String input) {
+        String[] parts = input.split(" ");
+        String desc = String.join(" ", Arrays.copyOfRange(parts, 1, parts.length));
+        String command = parts[0];
+        return new String[]{command, desc};
+    }
+
+    public static String[] processFileLine(String input) throws HsienException{
+        String[] parts = input.split(" ");
+        System.out.println(input);
+        String regex = "\\[(.*?)\\] \\[(.*?)\\] (.*)";
+        Pattern pattern = Pattern.compile(regex);
+        Matcher matcher = pattern.matcher(input.trim());
+
+        if (matcher.matches()) {
+            String command = matcher.group(1); // Extract command from the first group
+            String isMark = matcher.group(2); // Extract mark status from the second group
+            String desc = matcher.group(3); // Extract description from the third group
+            return new String[]{command, isMark, desc};
+        } else {
+            throw new HsienException();
+        }
+    }
+
+    public static void addTaskFromFile(String command, String isMark, String desc, ArrayList<Task> messages) {
+        Task newTask = null;
+        String tempDesc;
+
+        // Create Task object based on action
+        if (command.equals("T")) {
+            newTask = new Todo(desc);
+        } else if (command.equals("D")) {
+            tempDesc = desc.split("\\(by:")[0].trim();
+            String byDate = desc.split("\\(by:")[1].trim();
+            newTask = new Deadline(tempDesc, byDate);
+        } else {
+            tempDesc = desc.split("\\(from:")[0].trim();
+            String[] dates = desc.split("\\(from:")[1].split("to:");
+            String fromDate = dates[0].trim();
+            String toDate = dates[1].substring(0, dates[1].length()-1).trim();
+            newTask = new Event(tempDesc, fromDate, toDate);
```

Comment:
Consider using a switch case block instead of a if-else, as the values to expect are predefined.